lib/tasks/valid_test_data_generator.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require "tasks/school_urn_generator"
+require "tasks/trn_generator"
+
+module ValidTestDataGenerator
+  class LeadProviderPopulator
+    class << self
+      def call(name:, total_schools: 10, participants_per_school: 100)
+        new(name: name).call(total_schools: total_schools, participants_per_school: participants_per_school)
+      end
+    end
+
+    def call(total_schools: 10, participants_per_school: 100)
+      generate_new_schools(count: total_schools - lead_provider.schools.count) if lead_provider.schools.count < total_schools
+      lead_provider.schools.order("created_at desc").limit(total_schools).each do |school|
+        find_or_create_participants(school: school, number_of_participants: participants_per_school)
+      end
+    end
+
+  private
+
+    attr_reader :lead_provider
+
+    def initialize(name:)
+      @lead_provider = ::LeadProvider.find_or_create_by!(name: name)
+    end
+
+    def generate_new_schools(count:)
+      count.times { create_fip_school_with_cohort(urn: SchoolURNGenerator.next) }
+    end
+
+    def find_or_create_participants(school:, number_of_participants:)
+      generate_new_participants(school: school, count: number_of_participants - school.ecf_participants.count) if school.ecf_participants.count < number_of_participants
+      school.ecf_participants.order("created_at desc").limit(number_of_participants).to_a
```

Comment:
What is this last line doing? Nothing that we use from what I can see.

Reply:
Returning an array of the chosen schools. I was mapping them before, but you're correct in that it's now a redundant line.